graphql_query_derive/src/lib.rs

```diff
@@ -26,38 +27,48 @@ fn graphql_query_derive_inner(
 ) -> Result<proc_macro::TokenStream, syn::Error> {
     let input = TokenStream::from(input);
     let ast = syn::parse2(input)?;
-    let (query_path, schema_path) = build_query_and_schema_path(&ast)?;
-    let options = build_graphql_client_derive_options(&ast, query_path.clone())?;
-
-    generate_module_token_stream(query_path, &schema_path, options)
-        .map(Into::into)
-        .map_err(|err| {
-            syn::Error::new_spanned(
-                ast,
-                format!("Failed to generate GraphQLQuery impl: {}", err),
-            )
-        })
-}
 
-fn build_query_and_schema_path(input: &syn::DeriveInput) -> Result<(PathBuf, PathBuf), syn::Error> {
     let cargo_manifest_dir = env::var("CARGO_MANIFEST_DIR").map_err(|_err| {
         syn::Error::new_spanned(
-            input,
+            &ast,
             "Error checking that the CARGO_MANIFEST_DIR env variable is defined.",
         )
     })?;
 
-    let query_path = attributes::extract_attr(input, "query_path")?;
-    let query_path = format!("{}/{}", cargo_manifest_dir, query_path);
-    let query_path = Path::new(&query_path).to_path_buf();
-    let schema_path = attributes::extract_attr(input, "schema_path")?;
+    let schema_path = attributes::extract_attr(&ast, "schema_path")?;
     let schema_path = Path::new(&cargo_manifest_dir).join(schema_path);
-    Ok((query_path, schema_path))
+
+    let query_string = attributes::extract_attr(&ast, "query_string").ok();
```

Comment:
Maybe just `query` (as the default or only supported method of providing the query)?

```suggestion
    let query_string = attributes::extract_attr(&ast, "query").ok();
```

Reply:
I’d vote for not shipping the breaking change just yet, so I think `query_string` for now. If there is a plan for a major release, we can do the refactor for that release.